docs/advanced-topics/dockstore-cli/dockstore-cli-faq.rst

```diff
@@ -0,0 +1,91 @@
+    env variable) to setup volumes with docker, which on a Mac can default
+    to a subdirectory of /var. In order to get ``cwltool`` working on your
+    Mac, you need to set your TMPDIR to be under one of the shared volumes
+    in Docker for Mac. You can do this by doing something similar to the
+    following:
+    ::
+
+        export TMPDIR=/tmp/docker_tmp
+
+By default, Docker for Mac allocates fewer resources (CPU, Memory, Swap)
+to containers compared to what is available on your host machine. You
+can change what it allocates using the Docker for Mac GUI under
+``Preferences > Advanced`` as described
+`here <https://docs.docker.com/docker-for-mac/#advanced>`__.
+
+* The default allocation can cause workflows or tools to fail without informing the user with a memory or resource related error message. If you find that your workflow or tool is behaving differently on a Mac compared to a similarly resourced Ubuntu environment, you can try increasing the resources allocated to Docker on the Mac to resolve the discrepancy.
+
+How do I launch tools/workflows without internet access on compute nodes?
+-------------------------------------------------------------------------
+
+Some tools/workflows require Docker images to launch even if they are
+local entries. If the compute nodes do not have internet access, you can
+follow these steps:
+
+1. download the Docker image(s) on the head node which does have internet access using the ``docker save -o <filename> <imagename>``
+2. ensure that the ``<imagename>`` matches the image name specific in the CWL or WDL descriptor 
+3. place the image file(s) in a location that the compute nodes have access to (make sure there are only images in that directory)
+4. specify in the dockstore config file (default ~/.dockstore/config) the directory that contains your image(s) using ``docker-images = /home/user/docker_images_directory``
+
+The Dockstore CLI will automatically load all Docker images in the
+directory specified prior to a ``launch --local-entry`` command.
+
+.. _cromwell-docker-lockup:
+
+I was running a WDL locally, but some of my tasks are failing randomly and/or now I cannot use Docker.
+------------------------------------------------------------------------------------------------------
+
+This is a known issue with how Cromwell and Docker, which the Dockstore CLI use to launch WDL workflows, manage resources on a local machine. Certain problems related to resource management may happen when running locally that do not happen when running on the cloud. These problems are much more likely to happen if you are running a computationally intensive scattered task, such as LD pruning 23 chromosomes where each chromosome is an instance of a scattered task. The two most common problems we see are a "Docker lockup" and stochastic failure of tasks. It is possible for one, both, or neither of these problems to occur during a single submission.
```

Comment:
This one would be use instead of uses because the subject is plural. 

Docker manages resources on a local machine, and by extension Cromwell does too because Cromwell is the one that decides when to launch Docker containers, as per concurrent-job-limit.

Reply:
1. The subject for the phrase (clause?, my formal grammar is weak) is "Dockstore CLI", so it's singular. Try substituting a singular personal pronoun as proof: "This is a known issue with how Cromwell and Docker, which Charles use to launch WDL" -- it has to be "uses" there.
2. I guess it's how you interpret "managing resources". If I do a `docker run...` or `docker-compose...up` from the command line or from a bash script, am I managing resources? I guess so. I tend to interpret it as more of a low-level thing, but I can see it your way.

Reply:
You has convinced me!